### DISCUSSION TITLE
July 16 - Literate Programming IV - publishing

### DISCUSSION THREAD

**Discussion by tonyfast** [Quansight/writers-workshop#23] [New class]
# the goal of the workshop

[what to do with your mess of writing? publish?](https://gist.github.com/tonyfast/c040b39d6149ef4fa57343f073dbde38)

## so far

1. introduction to literate programs where code and narrative contribute equally to the narrative.
2. introduction to different media formats in the notebook and their use as collage.
3. introduction to testing a mess of notebooks.

## teacher's preferred ground rules

* How to use the raise handle feature? Keeping the professor on track..
* I encourage y'all to interrupt me when I'm talking or demo'ing. 
  Sometimes I get distracted by my own special interests and this class if for you all.
* To help take notes, please respond to this discussion item and continue editing your one reply during class. i can see the changes on the discussion board as they happen.

## [calls to action] :+1: :-1:  :smile: :confused:  :rocket:  :heart:  :eyes: :tada: attendance

* place any questions you have in the discussions and we can try to discuss them during the workshop.

* react to this discussion so we can take attendance.

![image](https://user-images.githubusercontent.com/4236275/117461813-fd163180-af1b-11eb-901b-bfdee65d8bdc.png)

### Q1 how's your energy

* :+1: i feel energetic from this week
* :-1: i feel exhausted from this week

## note taking

please use this discussion as a place to take notes. create a comment that you edit throughout the lesson. 

_the perfect is the enemy of the good_

## agenda

0. [:00-:07] chit chat
1. show and tell
2. hands on publishing
3. how would collect and curate your works?
4. homeworks
5. [:50-:60] wrapup

write! right?

**Comment by isabela-pf** [comment]
Notes:

> *Restart and run all or it didn't happen*

A request: write about napping


[publishing notebooks](https://gist.github.com/tonyfast/c040b39d6149ef4fa57343f073dbde38) (a notebook that is published.)
Trivia time:
1. What was the subject of the first movie ever made in 1878? a man riding a horse
2.
3. What else id Guttenberg credited with inventing that enabled the print revolution? moveable type
4. Longest running scientific journal? The Philosophical Transactions of the Royal Society

Don't let the infrastructure around where the writing will be published get in the way of writing. Let's make documentation simple.

We joined on a binder and used JLab 3's RTC!
- [jupyter book](https://jupyterbook.org/intro.html)
- [gist](https://gist.github.com/) is just file sharing (what a revelation)
- [nox](https://www.noxlibrary.org/)
- [doit](https://pydoit.org/)

Full documentation just sitting in a single file in JupyterLab. Just a whole searchable website in a readme. Nothing special. NO. That is a lie; this is amazing!

Where's the documentation for this documentation 👀 ???

**Comment by MarsBarLee** [comment]
Notes:
Trivia: What is the name of the longest running scientific journal that was started by the Royal Society in 1660? -> apparently the answer is NOT '**alchemy for dummies**'

**Doit & Nox**
import doit, nox
nox is a programmatic way to write virtual environemnts and conda environments
doit is a programmatic way that make files on disc, think make in python
%reload_ext doit
from mee.tasks import *
!nox -s user -- you-gh-name

write gist and cofigure best way to write docs?
github profile readmd.md -> daily updates your readme with the gist
a sphinx build is somewhere here and if we're lucky, a pdf
decentralize, write is gist instead of repo. then make searchable doc

learning to aggregate our work

your readme/doc is the doorhandle- influences how people enter and interact with your work

docs is just configuration files

**Comment by willwray** [comment]
Calling my next library trivium

[`doit`](https://pydoit.org/)  programmatic way to make files on disc, think make in python
%reload_ext doit
import nox         programmatic virtual / conda environments
!nox -s user -- github-username
!nox -s sphinx -- github-username
!nox -s sphinx -- jtpio
!nox -s mkdocs -- jtpio
